cdap-ui/app/cdap/components/SystemServicesDelay/index.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as React from 'react';
+import { connect, Provider } from 'react-redux';
+import SystemDelayStore from 'services/SystemDelayStore';
+import SystemDelayActions from 'services/SystemDelayStore/SystemDelayActions';
+import Snackbar from '@material-ui/core/Snackbar';
+import Button from '@material-ui/core/Button';
+import ee from 'event-emitter';
+import { WINDOW_ON_FOCUS, WINDOW_ON_BLUR } from 'services/WindowManager';
+import { getExperimentValue, isExperimentEnabled } from 'services/helpers';
+import DataSource from 'services/datasource';
+import cloneDeep from 'lodash/cloneDeep';
+import flatten from 'lodash/flatten';
+
+interface IBinding {
+  resource: {
+    id: string;
+    requestTime: number;
+  };
+  type: string;
+  checkAgainsLeft?: number;
+}
+
+interface ISystemDelayProps {
+  showDelay: boolean;
+  activeDataSources: DataSource[];
+}
+
+const EXPERIMENT_ID = 'system-delay-notification';
+const HEALTH_CHECK_INTERVAL = 12000;
+const DEFAULT_DELAY_TIME = 5000;
+
+class SystemServicesDelayView extends React.Component<ISystemDelayProps> {
+  public state = { delayedBindings: [] };
```

Comment:
It looks like delayedBindings should be an object?

Reply:
fixed